packages/components/src/edit-in-place-control/index.js

```diff
@@ -0,0 +1,102 @@
+/**
+ * External dependencies
+ */
+import { noop, negate, isUndefined } from 'lodash';
+
+/**
+ * WordPress dependencies
+ */
+import { useState, useRef, useEffect } from '@wordpress/element';
+/**
+ * Internal dependencies
+ */
+import Button from '../button';
+
+const baseInputCssClass = 'components-edit-in-place-control__input';
+const beforeTransitionInputCssClass = 'small';
+const afterTransitionInputCssClass = 'large';
+const withoutTransitionCssClass = `${ baseInputCssClass } ${ beforeTransitionInputCssClass }`;
+const withTransitionCssClass = `${ baseInputCssClass } ${ afterTransitionInputCssClass }`;
+const cancelEvent = ( event ) => (
+	event.preventDefault(), event.stopPropagation()
+);
+
+function EditInPlaceControl( {
+	initialValue = '',
+	onClick = noop,
+	onUpdate = noop,
+	switchToEditModeButtonLabel,
+	inputValidator = negate( isUndefined ),
+	editNameInputLabel = initialValue,
+} ) {
+	const [ isEdit, setIsEdit ] = useState( false );
+	const [ value, setValue ] = useState( initialValue );
+	const [ inputCssClasses, setInputCssClasses ] = useState(
+		withoutTransitionCssClass
+	);
+
+	const inputRef = useRef();
+	const buttonRef = useRef();
+
+	useEffect( () => {
+		if ( isEdit ) {
+			inputRef.current.focus();
+			inputRef.current.select();
+		} else {
+			buttonRef.current.focus();
+		}
+	}, [ isEdit ] );
+
+	return (
+		<>
+			{ isEdit ? (
+				<input
```

Comment:
Similar to the issue with the `Button`, this also causes a warning. It would have to be implemented as a ToolbarItem.

Reply:
altered to `<ToolbarItem as="input"`